semgrep-core/src/osemgrep/cli_scan/Output.ml

```diff
@@ -13,256 +13,11 @@ module Out = Semgrep_output_v0_j
-      in
-      (*  need to prefix with the dotted path of the config file *)
-      let check_id = env.config_prefix ^ rule_id in
-      let metavars = Some metavars in
-      (* LATER: this should be a variant in semgrep_output_v0.atd
-       * and merged with Constants.rule_severity
-       *)
-      let severity = string_of_severity rule.severity in
-      let metadata =
-        match rule.metadata with
-        | None -> `Assoc []
-        | Some json -> JSON.to_yojson json
-      in
-      let lines = lines_of_file (start, end_) path |> String.concat "\n" in
-      {
-        check_id;
-        path;
-        start;
-        end_;
-        extra =
-          {
-            metavars;
-            lines;
-            (* fields derived from the rule *)
-            message;
-            severity;
-            metadata;
-            (* TODO: other fields derived from the rule *)
-            fix;
-            fix_regex = None;
-            (* TODO: extra fields *)
-            is_ignored = Some false;
-            (* LATER *)
-            (* TODO: rule_match.match_based_id *)
-            fingerprint = "0x42";
-            sca_info = None;
-            fixed_lines = None;
-            dataflow_trace = None;
-          };
-      }
-
-let cli_output_of_core_results (conf : Scan_CLI.conf) (res : Core_runner.result)
-    : Out.cli_output =
-  match res.core with
-  | {
-   matches;
-   errors;
-   (* LATER *)
-   skipped_targets = _;
-   skipped_rules = _;
-   explanations = _;
-   stats = _;
-   time = _;
-  } ->
-      let env =
-        { hrules = res.hrules; config_prefix = config_prefix_of_conf conf }
-      in
-      (* TODO: not sure how it's sorted. Look at rule_match.py keys? *)
-      let matches =
-        matches
-        |> List.sort (fun (a : Out.core_match) (b : Out.core_match) ->
-               compare a.rule_id b.rule_id)
-      in
-      (* TODO: not sure how it's sorted, but Set_.elements return
-       * elements in OCaml compare order (=~ lexicographic for strings)
-       *)
-      let scanned = res.scanned |> Set_.elements in
-      {
-        version = Some Version.version;
-        results = matches |> Common.map (cli_match_of_core_match env);
-        paths =
-          {
-            scanned;
-            _comment = Some "<add --verbose for a list of skipped paths>";
-            (* TOPORT *)
-            skipped = None;
-          };
-        (* TODO: *)
-        errors = errors |> Common.map cli_error_of_core_error;
-        (* LATER *)
-        time = None;
-        explanations = None;
-      }
-
 (*****************************************************************************)
 (* Entry point *)
 (*****************************************************************************)
 
-let output_result (conf : Scan_CLI.conf) (res : Core_runner.result) : unit =
-  let (cli_output : Out.cli_output) = cli_output_of_core_results conf res in
+let output_result (conf : Scan_CLI.conf) (cli_output : Out.cli_output) : unit =
```

Comment:
I'm not sure we want to change the type of output_result. It was fine getting Core_runner.result because there might
be more information in it at some point than in Out.cli_output.